server_admin/topics/clients/client-policies.adoc

```diff
@@ -0,0 +1,159 @@
+some executor checks whether the value of the parameter `redirect_uri` in the authorization request matches exactly with one of the pre-registered redirect URIs on
+Authorization Endpoint and rejects this request if not.
+
+The executor can not be used solely by itself. It can be used in a <<_client_policy_profile,profile>> that is described afterwards.
+
+An executor has it's own SPI. Developers can implement their own executor by implementing the particular provider, however SPI is considered private and can change in the future.
+
+An executor can be configurable the same as other configurable provider. What can be configured depends on the nature of each executor.
+
+An executor acts on various events. An executor implementation can ignore certain types of events (For example executor for checking OIDC `request` object acts just
+on the OIDC authorization request). Events are:
+
+* Creating a client (including creation through dynamic client registration)
+* Updating a client
+* Sending an authorization request
+* Sending a token request
+* Sending a token refresh request
+* Sending a token revocation request
+* Sending a token introspection request
+* Sending a userinfo request
+* Sending a logout request with a refresh token
+
+On each event, an executor can work in multiple phases. For example, on creating/updating a client, the executor can modify the client configuration by auto-configure specific client
+settings. After that, the executor validate this configuration in validation phase.
+
+One of several purposes for this executor is to realize security requirements of client conformance profiles like FAPI. To do so, following executors are needed:
+
+* Enforce secure <<_client-credentials,Client Authentication method>> is used for the client
+* Enforce <<_mtls-client-certificate-bound-tokens,Holder-of-key tokens>> are used
+* Enforce <<_proof-key-for-code-exchange,Proof Key for Code Exchange (PKCE)>> is used
+* Enforce secure signature algorithm for <<_client-credentials,Signed JWT client authentication (private-key-jwt)>> is used
+* Enforce HTTPS redirect URI and make sure that configured redirect URI does not contain wildcards
+* Enforce OIDC `request` object satisfying high security level
+* Enforce Response Type of OIDC Hybrid Flow including ID Token used as _detached signature_ as described in the FAPI 1 specification, which means that ID Token returned from Authorization response won't contain user profile data
+* Enforce more secure `state` and `nonce` parameters treatment for preventing CSRF
+* Enforce more secure signature algorithm when client registration
+
+[[_client_policy_profile]]
+===== Profile
+
+A profile consists of several executors which can realize a security profile like FAPI. Profile can be configured by the Admin REST API (admin console) together with it's executors.
```

Comment:
Always use a comma before which
Never use a comma before that

Reply:
Great suggestion, Thanks